common/sgx/qeidentity.c

```diff
@@ -195,11 +218,34 @@ oe_result_t oe_enforce_qe_identity(sgx_report_body_t* qe_report_body)
             parsed_info.attributes_xfrm_mask,
             parsed_info.attributes.xfrm);
 
-    oe_cleanup_qe_identity_info_args(&qe_id_args);
+    if (qe_report_body->attributes.flags & SGX_FLAGS_DEBUG)
+        OE_RAISE_MSG(
+            OE_QUOTE_ENCLAVE_IDENTITY_VERIFICATION_FAILED,
+            "QE has SGX_FLAGS_DEBUG set!!",
+            NULL);
+
+    if (oe_datetime_compare(&parsed_info.issue_date, &from) > 0)
+        from = parsed_info.issue_date;
+    if (oe_datetime_compare(&parsed_info.next_update, &until) < 0)
+        until = parsed_info.next_update;
+
+    oe_datetime_log_info("QE identity overall issue date: ", &from);
+    oe_datetime_log_info("QE identity overall next update: ", &until);
+    if (oe_datetime_compare(&from, &until) > 0)
+        OE_RAISE_MSG(
+            OE_VERIFY_FAILED_TO_FIND_VALIDITY_PERIOD,
+            "Failed to find an overall QE identity validity period.",
+            NULL);
+
```

Comment:
Do you want to check validity_from and validity_until againt NULL before defrerencing them?

Reply:
Nice catch!